lib/src/solver/version_solver.dart

```diff
@@ -99,6 +99,9 @@ class VersionSolver {
 
       return await _result();
     } finally {
+      // Stop pre-fetching package/version listings from hosted repository, as
+      // resolution is done.
+      _systemCache.hosted.stopPrefetching();
```

Comment:
Hmm, are we building a foot-gun here?

I'm wondering if there's other places where we call `BoundHostedSource` and could accidentally be triggering a pre-fetch operation?

What if as a first step in the `try {` block above we initiate pre-fetching if everything that is unsatisfied, since this must be known at this point, right?

Then pre-fetching only happens when `_systemCache.hosted.startPrefetching(...)` has been called, and it always steops when `_systemCache.hosted.stopPrefetching()` is called. Well, I mean most importantly it is necer necessary to call `_systemCache.hosted.stopPrefetching()` if you haven't called `_systemCache.hosted.startPrefetching(...)`.

Thoughts? Or maybe pre-fetching is a zone thing? Just thinking here..

Reply:
> I'm wondering if there's other places where we call BoundHostedSource and could accidentally be triggering a pre-fetch operation?

Good point. I kind of like the zone idea.
We enter a pefetching-zone, and the fetching operation of BoundHostedSource only does prefetching when inside that zone.